### PR TITLE
[Feat] sendSmartTransaction with multiple signers

### PR DESCRIPTION
**Pull request by kAsky53**

### CODE REVIEW

File: src/RpcClient.ts [hunk 1]
```diff
@@ -506,17 +506,21 @@ export class RpcClient {
   /**
    * Build and send an optimized transaction, and handle its confirmation status
    * @param {TransactionInstruction[]} instructions - The transaction instructions
-   * @param {Keypair} fromKeypair - The sender's keypair
+   * @param {Signer[]} signers - The transaction signers. the first signer should be the fee payer
```
Comment (0xIchigo) [inline src/RpcClient.ts:509]: Nit: Can you rephrase this to `The transaction's signers. The first signer should be the fee payer`

File: src/RpcClient.ts [hunk 2]
```diff
@@ -506,17 +506,21 @@ export class RpcClient {
...
   */
   async sendSmartTransaction(
     instructions: TransactionInstruction[],
-    fromKeypair: Keypair,
+    signers: Signer[],
     lookupTables: AddressLookupTableAccount[] = [],
     sendOptions: SendOptions = { skipPreflight: false },
   ): Promise<TransactionSignature> {
+    if (!signers.length) {
+      throw new Error("Fee payer must sign the transaction");
```
Comment (0xIchigo) [inline src/RpcClient.ts:521]: Nit: Can you rephrase to `The fee payer must sign the transaction`

**Review by 0xIchigo — CHANGES_REQUESTED** [review]
Very based! I requested a few minor changes but it looks good to go! Would you be able to update [the example `sendSmartTransaction` in the README](https://github.com/helius-labs/helius-sdk?tab=readme-ov-file#sendsmarttransaction) to reflect these new changes, please?

**Comment by kAsky53** [comment]
@0xIchigo I've resolved all of your comments!

File: src/RpcClient.ts [hunk 3]
```diff
@@ -631,7 +640,11 @@ export class RpcClient {
 
           return await this.pollTransactionConfirmation(txtSig);
         } catch (error) {
-          continue;
+          if (error instanceof TransactionExpiredTimeoutError) {
+            continue;
+          } else {
+            throw error;
```
Comment (kAsky53) [inline src/RpcClient.ts:646]: @0xIchigo I think we should throw custom error messages if it's not an expired timeout one. What do you think?

File: src/RpcClient.ts [hunk 4]
```diff
@@ -577,16 +584,15 @@ export class RpcClient {
...
         options: { 
           recommended: true,
         },
       });
 
-      let priorityFeeRecommendation = priorityFeeResponse.priorityFeeEstimate || 0;
-      let microlamportsPerCU = Math.max(
-        priorityFeeRecommendation,
+      const microlamportsPerCU = Math.max(
+        priorityFeeEstimate ? Math.ceil(priorityFeeEstimate) : 0,
```
Comment (kAsky53) [inline src/RpcClient.ts:595]: @0xIchigo If `priorityFeeEstimate` is not integer it will throw error, so I rounded it up here.

File: src/RpcClient.ts (hunk 3 above)
Comment (0xIchigo) [reply src/RpcClient.ts:645]: I like this, but for now, I think we should just continue

Comment (0xIchigo) [reply src/RpcClient.ts:645]: I have to do some work for `sendSmartTransaction` so I'll merge as is, instead of waiting for this to be changed, and I'll update it myself

**Review by 0xIchigo — APPROVED** [review]
LGTM 🔥
